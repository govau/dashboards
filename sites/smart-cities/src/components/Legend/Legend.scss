@import '~style/main';

.legendItem {
  margin-left: 24px;

  > * {
    vertical-align: middle;
  }
}

.legendItemText {
  padding-left: 8px;
  font-size: 12px;
  font-weight: 500;
}
