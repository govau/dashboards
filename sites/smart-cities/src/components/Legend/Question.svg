<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <g fill-rule="nonzero" fill="#1077B5">
    <path d="M10,0 C4.48544186,0 0,4.48541395 0,10.000007 C0,15.5145884 4.48544186,20.000007 10,20.000007 C15.5146047,20.000007 20,15.5145884 20,10.000007 C20,4.4854186 15.5146047,0 10,0 Z M10,1.39534884 C14.7605116,1.39534884 18.6046512,5.23951628 18.6046512,10.000007 C18.6046512,14.7604953 14.7605116,18.6046581 10,18.6046581 C5.23953488,18.6046581 1.39534884,14.7604721 1.39534884,10.000007 C1.39534884,5.23952093 5.23953488,1.39534884 10,1.39534884 Z M10,3.48837209 C8.0535814,3.48837209 6.46313953,4.75618605 6.06832558,6.3372093 C5.99597394,6.58258472 6.06368398,6.84788535 6.24477165,7.02855808 C6.42585933,7.20923081 6.69128411,7.27630167 6.93644398,7.2033385 C7.18160385,7.13037534 7.36718319,6.92907905 7.42004651,6.67877907 C7.63209302,5.82982558 8.6874186,4.88372093 10,4.88372093 C10.6466047,4.88372093 11.3238372,5.12116279 11.8023256,5.50872093 C12.2808372,5.89627907 12.5581395,6.39534884 12.5581395,6.97674419 C12.5581395,7.74030233 12.4026047,8.08305814 12.1947674,8.3502907 C11.9869535,8.61754651 11.6546977,8.82531395 11.2209302,9.13517442 C10.7871628,9.44505814 10.2719302,9.8728907 9.89825581,10.5668674 C9.52465116,11.2608442 9.30232558,12.1899372 9.30232558,13.4883791 C9.3023256,13.8737532 9.61468507,14.1861607 10,14.1861607 C10.3853149,14.1861607 10.6976744,13.8737532 10.6976744,13.4883791 C10.6976744,12.3449605 10.8896977,11.6679837 11.1264651,11.228193 C11.3632093,10.7884721 11.654814,10.5476814 12.0348837,10.2761698 C12.4149302,10.0046581 12.8968837,9.72547209 13.2994186,9.20784884 C13.7019767,8.69022093 13.9534884,7.95737209 13.9534884,6.97674419 C13.9534884,5.93023256 13.416186,5.0267093 12.6744186,4.42587209 C11.9326047,3.82505814 10.979186,3.48837209 10,3.48837209 L10,3.48837209 Z M10.0072093,14.6511698 C9.49304651,14.6511698 9.06972093,15.0676581 9.06972093,15.5814023 C9.06972093,16.0951465 9.49306977,16.5116349 10.0072093,16.5116349 C10.5213488,16.5116349 10.9374419,16.0951465 10.9374419,15.5814023 C10.9374419,15.0676581 10.5213256,14.6511698 10.0072093,14.6511698 Z"></path>
  </g>
</svg>
