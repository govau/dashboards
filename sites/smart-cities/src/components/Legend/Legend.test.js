import React from 'react';
import { shallow } from 'enzyme';
import Legend from './Legend';

const defaultProps = {
  series: [
    {
      color: 'red',
      name: 'House'
    },
    {
      color: 'blue',
      name: 'Apartment'
    },
    {
      color: 'grey',
      name: 'Other'
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <Legend {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render the correct number of legend items', () => {
  const component = shallow(
    <Legend {...defaultProps} />
  );

  expect(component.find('.legendItem').length).toBe(defaultProps.series.length);
});
