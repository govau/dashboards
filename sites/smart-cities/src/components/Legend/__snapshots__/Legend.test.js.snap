// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className={[undefined]}>
  <div className=\\"legendItem\\">
    <svg width=\\"20\\" height=\\"20\\">
      <circle cx=\\"10\\" cy=\\"10\\" r=\\"10\\" fill=\\"red\\" />
    </svg>
    <span className=\\"legendItemText\\">
      House
    </span>
  </div>
  <div className=\\"legendItem\\">
    <svg width=\\"20\\" height=\\"20\\">
      <circle cx=\\"10\\" cy=\\"10\\" r=\\"10\\" fill=\\"blue\\" />
    </svg>
    <span className=\\"legendItemText\\">
      Apartment
    </span>
  </div>
  <div className=\\"legendItem\\">
    <svg width=\\"20\\" height=\\"20\\">
      <circle cx=\\"10\\" cy=\\"10\\" r=\\"10\\" fill=\\"grey\\" />
    </svg>
    <span className=\\"legendItemText\\">
      Other
    </span>
  </div>
</div>"
`;
